src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/BitcoinHeaderVersionRule.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Base.Deployments;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>
+    /// Checks if <see cref="BitcoinMain"/> network block's header has a valid block version.
+    /// </summary>
+    [HeaderValidationRule(CanSkipValidation = true)]
+    public class BitcoinHeaderVersionRule : HeaderVersionRule
+    {
+        /// <inheritdoc />
+        /// <exception cref="ConsensusErrors.BadVersion">Thrown if block's version is outdated or otherwise invalid.</exception>
+        public override Task RunAsync(RuleContext context)
+        {
```

Comment:
Method lacks entry and exit logs.

Reply:
I checked several other consensus rules and the standard appears to be that exit logs are only used when rules fail validation. Entry logs are not used at all, and the helper methods used within the rules have no entry/exit logs at all.

Reply:
Yeah, logging style is violated in other rules too. However that doesn't mean that we shouldn't introduce logs